Automattic/wp-calypso
# SegmentedControl to packages (Step 2): Temporarily shim old Calypso SegmentedControl component

Related to https://github.com/Automattic/wp-calypso/issues/81117
Branched from https://github.com/Automattic/wp-calypso/pull/84098

## Proposed Changes

- We point the old `SegmentedControl` component to the new one in `@automattic/components`, allowing us to transition from the old code to the new in smaller pieces and without risking updates to the old component in the interim.
- Remove old docs and files

## Testing Instructions



- Ensure `/devdocs` renders the `SegmentedControl` fine
- Test instances of the `SegmentedControl` and `SimplifiedSegmentedControl` components in Calypso. Examples:

    - `/stats/insights/[site]` (SimplifiedSegmentedControl):

    <img width="400" alt="Screenshot 2023-11-10 at 12 49 10 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/23369854-817d-435c-a544-fec3ef410150">

    - `/posts/[site]`:

    <img width="400" alt="Screenshot 2023-11-10 at 1 04 48 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/86ebc1ce-ad29-44d9-a0c0-4c46351ae414">

    - `/start/plans`:

    <img width="400" alt="Screenshot 2023-11-10 at 1 07 59 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/76b51551-4dc9-40bd-98a1-0f6916717e28">


## Pre-merge Checklist
